daemon/container.go

```diff
@@ -896,21 +896,21 @@ func (container *Container) Copy(resource string) (io.ReadCloser, error) {
 		return nil, err
 	}
 
-	basePath, err := container.getResourcePath(resource)
-	if err != nil {
+	if err := container.mountVolumes(); err != nil {
 		container.Unmount()
```

Comment:
Yes, this function is returning an io stream, if we defer the unmounts, they will be unmounted before the tar is created/read.